packages/node/src/policy-gen/generate.js

```diff
@@ -40,65 +67,107 @@ import { compartmentMapToPolicy } from './to-policy.js'
  *
- * @param {string | URL} entrypointPath
+ * @param {string | URL} entrypoint
  * @param {GenerateOptions} [opts]
- * @returns {Promise<LavaMoatPolicy>}
+ * @returns {Promise<{
+ *   policy: LavaMoatPolicy
+ *   compartmentMap: CompartmentMapDescriptor
+ * }>}
  */
 const generate = async (
-  entrypointPath,
+  entrypoint,
   {
     readPowers = defaultReadPowers,
     debug = false,
     policyOverride,
     isBuiltin,
     log = defaultLog,
     dev = false,
+    trustRoot = DEFAULT_TRUST_ROOT_COMPARTMENT,
     ...archiveOpts
   } = {}
 ) => {
   const conditions = devToConditions(dev)
 
-  /** @type {LoadCompartmentMapOptions} */
-  const loadCompartmentMapOptions = {
-    ...archiveOpts,
-    log,
-    readPowers,
-    conditions,
-    policyOverride,
-  }
+  log.debug('Loading compartment map…')
   const { compartmentMap, sources, renames } = await loadCompartmentMap(
-    entrypointPath,
-    loadCompartmentMapOptions
+    entrypoint,
+    {
+      ...archiveOpts,
+      log,
+      readPowers,
+      conditions,
+      policyOverride,
+      trustRoot,
+    }
   )
 
   /** @type {CompartmentMapToPolicyOptions} */
-  const baseOpts = { readPowers, policyOverride, isBuiltin, log }
+  const baseOpts = {
+    readPowers,
+    policyOverride,
+    isBuiltin,
+    log,
+    trustRoot,
+  }
 
   // this weird thing is to make TS happy about the overload
   const opts = debug ? { debug: true, ...baseOpts } : baseOpts
 
-  return compartmentMapToPolicy(compartmentMap, sources, renames, opts)
+  const policy = compartmentMapToPolicy(
+    entrypoint,
+    compartmentMap,
+    sources,
+    renames,
+    opts
+  )
+  return { policy, compartmentMap }
 }
 
 /**
- * Returns `true` if we should write a debug policy
+ * Reports policy override resources which weren't found on disk and are thus
+ * not in the compartment map descriptor.
  *
- * @param {Pick<GeneratePolicyOptions, 'write' | 'debug'>} [opts]
- * @returns {boolean}
+ * @param {CompartmentMapDescriptor} compartmentMap
+ * @param {ReportInvalidOverridesOptions} options
+ * @returns {void}
  */
-const shouldWriteDebugPolicy = (opts = {}) => !!(opts.write && opts.debug)
+const reportInvalidOverrides = (
+  compartmentMap,
+  { policyOverride, policyOverridePath, log = defaultLog }
+) => {
+  if (!policyOverride) {
+    return
+  }
 
-/**
- * Absolute path to the default policy debug file
- */
-const ABS_POLICY_DEBUG_PATH = nodePath.resolve(DEFAULT_POLICY_DEBUG_PATH)
+  const canonicalNames = new Set(
+    values(compartmentMap.compartments).map(({ label }) => label)
+  )
+  // TODO: use `Set.prototype.difference` once widely available
+  const invalidOverrides = keys(policyOverride.resources).filter(
+    (key) => !canonicalNames.has(key)
+  )
+
+  if (invalidOverrides.length) {
+    let msg = `The following resource(s) provided in policy overrides`
+    msg += policyOverridePath ? ` (${hrPath(policyOverridePath)})` : ''
+    msg += ` were not found and may be invalid:\n`
```

Comment:
Not found is ambiguous in meaning I think. More precisely - could not be matched with corresponding location in node_modules

Reply:
yeah, we can improve it later if so desired.